### ISSUE TITLE
[Steamdeck] Unable to launch games. Tested Salt and Sanctuary and XCOM2.

### ISSUE THREAD

**Issue by klin0503** [Heroic-Games-Launcher/HeroicGamesLauncher#1378]
Edit: changed to proper format.

**Describe the bug**
When I try to launch Salt and Sanctuary, nothing really happens, game doesn't boot up. There is a Steam loading icon, then it will just go back to the game page with the "Play Now" button. When I try to run XCOM2, I get a similar error, except the last line says '/lib/ld-linux.so.2: could not open'. I have only tested these 2 games so far.

**Add logs**
See below.

**Steps to reproduce**
Launch a game in game mode using heroic launcher.

**Screenshots**
N/A

**System Information**
OS SteamOS
Heroic Version Flatpak 2.3.3

**Additional information**
Games are installed on SD card. Games are from Epic game store.

```
Launch Command: STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH=/home/deck/Games/Heroic/Prefixes/SaltAndSanctuary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 499f085e442048aa854d6250b21c42fb --no-wine --wrapper "/home/deck/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier/_v2-entry-point -- '/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton' waitforexitandrun"

System Info:

  Heroic Version: 2.3.3 Brook
  Legendary Version:  0.20.25 Our Benefactors
  OS: Freedesktop.org KERNEL: 5.13.0-valve10.3-1-neptune-02176-g5fe416c4acd8 ARCH: x64
  CPU: AMD Custom APU 0405 @1.76 GOVERNOR: schedutil
  RAM: Total: 14.49 GiB Available: 11.9 GiB
  GRAPHICS: GPU0: Device 163f VRAM: 512MB DRIVER:  
  PROTOCOL: wayland
 

Game Settings: {
"checkUpdatesInterval": 10,
"enableUpdates": false,
"addDesktopShortcuts": false,
"addStartMenuShortcuts": false,
"autoInstallDxvk": false,
"autoInstallVkd3d": false,
"checkForUpdatesOnStartup": false,
"customWinePaths": [],
"defaultInstallPath": "/run/media/mmcblk0p1/Heroic",
"defaultWinePrefix": "/home/deck/Games/Heroic/Prefixes",
"language": "en",
"maxWorkers": 0,
"minimizeOnLaunch": true,
"nvidiaPrime": false,
"otherOptions": "",
"showUnrealMarket": false,
"showFps": false,
"useGameMode": false,
"userInfo": {
"epicId": "c171d6a563484c07ba8c307f1e60a8e9",
"name": "deck"
},
"wineCrossoverBottle": "Heroic",
"winePrefix": "/home/deck/Games/Heroic/Prefixes/SaltAndSanctuary",
"wineVersion": {
"bin": "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton",
"name": "Proton - Proton 7.0",
"type": "proton"
},
"altLegendaryBin": "",
"altGogdlBin": "",
"disableController": false,
"egsLinkedPath": "",
"maxRecentGames": 5,
"maxSharpness": 2,
"savesPath": "",
"targetExe": "",
"useSteamRuntime": true
}

Error Log:
[Core] WARNING: Could not determine locale, falling back to en-US
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching 499f085e442048aa854d6250b21c42fb...
pressure-vessel-wrap[224]: W: /dev/shm not shared between app instances (flatpak#4214). The Steam Overlay will not work.
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
[S_API FAIL] SteamAPI_Init() failed
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
terminate called after throwing an instance of 'dxvk::DxvkError'
```

**Comment by flavioislima** [comment]
try gong to the game settings > Other > Steam Runtime and disable it.
Salt & Sanctuary doesn't run on linux last time I checked. Never tried Xcom 2 though.

**Comment by klin0503** [comment]
Same behaviour, but different error. On protondb, there are no reports for steam deck, but it does say it runs natively on Linux.

[https://www.protondb.com/app/283640](url)

Error log after disabling Steam Runtime:
```
Launch Command: STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH=/home/deck/Games/Heroic/Prefixes/SaltAndSanctuary /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary launch 499f085e442048aa854d6250b21c42fb --no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton' run"

System Info:

  Heroic Version: 2.3.3 Brook
  Legendary Version:  0.20.25 Our Benefactors
  OS: Freedesktop.org KERNEL: 5.13.0-valve15-1-neptune-02197-gf6ec7ad3762a ARCH: x64
  CPU: AMD Custom APU 0405 @1.81 GOVERNOR: schedutil
  RAM: Total: 14.49 GiB Available: 12.19 GiB
  GRAPHICS: GPU0: Device 163f VRAM: 512MB DRIVER:  
  PROTOCOL: wayland
 

Game Settings: {
"autoInstallDxvk": false,
"autoInstallVkd3d": false,
"maxSharpness": 2,
"nvidiaPrime": false,
"otherOptions": "",
"savesPath": "",
"showFps": false,
"targetExe": "",
"useGameMode": false,
"wineCrossoverBottle": "Heroic",
"winePrefix": "/home/deck/Games/Heroic/Prefixes/SaltAndSanctuary",
"wineVersion": {
"bin": "/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton",
"name": "Proton - Proton 7.0",
"type": "proton"
},
"useSteamRuntime": false
}

Error Log:
[Core] WARNING: Could not determine locale, falling back to en-US
[cli] INFO: Logging in...
[Core] INFO: Trying to re-use existing login session...
[cli] INFO: Checking for updates...
[Core] INFO: Getting authentication token...
[cli] INFO: Launching 499f085e442048aa854d6250b21c42fb...
fsync: up and running.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
[S_API FAIL] SteamAPI_Init() failed
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
WARNING: radv is not a conformant Vulkan implementation, testing use only.
/lib/ld-linux.so.2: could not open
```

**Comment by klin0503** [comment]
Update: I reinstalled the heroic games launcher and now it works.
